bpf/bpf_lxc.c

```diff
@@ -173,24 +173,36 @@ static __always_inline int ipv6_l3_from_lxc(struct __ctx_buff *ctx,
 			   orig_dip.p4, *dstID);
 	}
 
+	/* When an endpoint connects to itself via service clusterIP, we need
+	 * to skip the policy enforcement. If we didn't, the user would have to
+	 * define policy rules to allow pods to talk to themselves. We still
+	 * want to execute the conntrack logic so that replies can be correctly
+	 * matched.
+	 */
+	if (hairpin_flow)
```

Comment:
I think we can do the `!defined(ENABLE_HOST_SERVICES_FULL) && !defined(DISABLE_LOOPBACK_LB)` approach here as well, skip a few instructions & make the ingress/egress cases more similar?

Reply:
I didn't add those because the compiler should already be able to compile this out automatically in such case (it should see that `hairpin_flow` is always false). I thought not having the code bloated with so many `#ifdef` would be better, but maybe it's best to have all the precondition clearly defined. I don't have a strong opinion either way.